### PR TITLE
User language

### PR DESCRIPTION
**Pull request by MarcoBuster**
Recently, **Telegram** has added `language_code` to [`User` object](https://core.telegram.org/bots/api#user) (for now it isn't in documentation).
In botogram framework, `User.language_code` is `user.lang`.

Example usage:
```python
REPLIES = {"en": "Hello! All right?", "it": "Ciao! Tutto bene?", "ru": "Здравствуйте! Хорошо?"}

@bot.command("start")
def start(chat, message):
    lang = message.sender.lang
    if lang not in REPLIES.keys():
        message.reply("Language not supported")
        return

    message.reply(REPLIES[lang])
```

### CODE REVIEW

File: docs/api/telegram.rst [hunk 1]
```diff
@@ -60,6 +60,12 @@ about its business.
 
       *This attribute can be None if it's not provided by Telegram.*
 
+   .. py:attribute:: lang
+
+      The user's language, given by Telegram.
+
+      *This attribute can be None if it's not provided by Telegram*
```
Comment (pietroalbini) [inline docs/api/telegram.rst]: You forgot to add the version in which this was added.

File: docs/api/telegram.rst [hunk 2]
```diff
@@ -60,6 +60,12 @@ about its business.
 
       *This attribute can be None if it's not provided by Telegram.*
 
+   .. py:attribute:: lang
+
+      The user's language, given by Telegram.
```
Comment (pietroalbini) [inline docs/api/telegram.rst]: Let's wait for the docs to see from where Telegram gets the language (client lang, os lang...).

File: docs/api/telegram.rst (hunk 1 above)
Comment (MarcoBuster) [inline docs/api/telegram.rst]: 0.4?

Comment (pietroalbini) [inline docs/api/telegram.rst]: Yep.

Comment (MarcoBuster) [inline docs/api/telegram.rst]: Done!

**Comment by MarcoBuster** [comment]
@pietroalbini Maybe we're ready to merge?

**Comment by pietroalbini** [comment]
Thinking a bit more about this, there can be a better API for this.

I'm thinking those attributes should be present:

* `User.language_code` with the full language code provided by Telegram
* `User.lang` with the language extracted with Babel (the `language` attribute of the Babel `Locale`)
* `User.country` with the country extracted with Babel (the `territory` attribute of the Babel `Locale`)

You can require [Babel](http://babel.pocoo.org) as a dependency, and then parse the language_code provided by Telegram with `Locale.parse`. The value should also be cached after being calculated.

**Comment by pietroalbini** [comment]
Is there any news on this?

File: docs/api/telegram.rst (hunk 2 above)
Comment (ALCC01) [reply docs/api/telegram.rst]: The upstream Telegram documentation now documents `language_code`, but does not specify where it comes from.

**Comment by MarcoBuster** [comment]
@pietroalbini Telegram doesn't seem to care about the user's country. `User.language_code` is **always** (in every client I tested) the same as `User.lang`, and `User.country` is always `None`. I propose not to use the integration with Babel and use only `User.lang` as unique identifier of user's language.

**Pull request merged**